### PR TITLE
Domains: Add DNSSEC management card

### PR DESCRIPTION
**Pull request by leonardost**
## Proposed Changes

This PR adds a DNSSEC management card to the domain management page.

For this first iteration, users will be able to toggle DNSSEC on and off. Also, when DNSSEC is enabled, they'll be able to view the `DNSKEY` and `DS` record values for their domain. This feature is currently only available for apex domains registered with WPCOM.

### Screenshots

- Placement of the DNSSEC card in the domain management page

Markup on 2024-07-11 at 16:00:30

- DNSSEC disabled

Screenshot 2024-07-11 at 16 01 02

- DNSSEC enabled

Screenshot 2024-07-16 at 18 43 11

## Why are these changes being made?

This is a long-standing user request that we are now able to handle (pMz3w-ich-p2).

## Testing Instructions

- This PR depends on D154951-code
- Open the live Calypso link or build this branch locally
- Go to the domain management page of a domain that was bought in WPCOM
- Ensure the DNSSEC card is present
- Try toggling DNSSEC on and off and ensure no error happens

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by delputnam** [comment]
@leonardost what do you think about just showing the DNSKEY and DS Record data in a <div> instead of an <input>.  This would let us show the entire records without scrolling or copying and pasting them.

File: client/my-sites/domains/domain-management/settings/cards/dnssec-card.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	};
+
+	const renderDsRecords = () => {
+		if ( ! dsData ) {
+			return null;
+		}
+
+		return (
+			<fieldset>
+				<legend>DS</legend>
```
Comment (delputnam) [inline client/my-sites/domains/domain-management/settings/cards/dnssec-card.tsx]: How about "DS Records" for the text here?

**Review by delputnam — APPROVED** [review]
Thanks for the updates.  I think this now works as expected. 👍

**Review by gius80 — APPROVED** [review]
LGTM! Tested alongside with D154951-code and works as expected.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16328469

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16328469).

**Pull request merged**